### DISCUSSION TITLE
Expose `Viewport.gui_accept_event()`

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#9853] [GUI]
`Control.accept_event()` calls `Viewport._gui_accept_event()`, this is handy for consuming the event inside a Control's script.

However, one would also want to `accept_event()` in a Control's `gui_input` signal. The signal is not necessarily handled by a script extending Control.

So the only way to consume the event in the signal handler is to get a Control node and call its `accept_event()` method.

I suggest exposing `Viewport._gui_accept_event()` to scripting. We've already exposed `Viewport.gui_release_focus()` for the same reason.

**Comment by theraot** [comment]
Looking at the source code, `_gui_accept_event` sets `gui.key_event_accepted = true;` and then calls `set_input_as_handled` (which is already exposed) if the `Viewport` is inside the scene tree. This makes me wonder if `set_input_as_handled` should do `gui.key_event_accepted = true;` and then `_gui_accept_event` could be removed.

**Reply by timothyqiu** [reply]
Yeah, I always wonder what's the difference between `set_input_as_handled()` and `accept_event()`.

**Reply by Cammymoop** [reply]
It seems as of 4.4+ `accept_event()` no longer sets `gui.key_event_accepted = true`. So it seems the only potential difference is that `Control.accept_event()` first checks `is_inside_tree()` before calling `get_viewport().set_input_as_handled()`.
So it looks like there's little to no difference at this point, though the documentation still leaves you second guessing without having to check the source unfortunately.

Edit: both the `Control` node and it's parent viewport check `is_inside_tree()` in turn actually.
